stan/math/prim/fun/quantile.hpp

```diff
@@ -0,0 +1,116 @@
+#ifndef STAN_MATH_PRIM_FUN_QUANTILE_HPP
+#define STAN_MATH_PRIM_FUN_QUANTILE_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/fun/as_array_or_scalar.hpp>
+#include <algorithm>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return sample quantiles corresponding to the given probabilities.
+ * The smallest observation corresponds to a probability of 0 and the largest to
+ * a probability of 1.
+ *
+ * Implements algorithm 7 from Hyndman, R. J. and Fan, Y.,
+ * Sample quantiles in Statistical Packages (R's default quantile function)
+ *
+ * @tparam T Type of input vector
+ * @param xs Numeric vector whose sample quantiles are wanted
```

Comment:
[optional] These names are fine though I tend to prefer more informative names like `dist_vec` or something like that since I then have to remember less when I'm reading where it's used